src/client/eppo-client.ts

```diff
@@ -393,16 +407,34 @@ export default class EppoClient implements IEppoClient {
     actions: Record<string, Attributes>, // TODO: ability to provide a set of actions with no context, or context broken out by numeric/categorical
     defaultValue: string,
   ): { variation: string; action: string | null } {
-    let variation = this.getStringAssignment(flagKey, subjectKey, subjectAttributes, defaultValue);
+    let variation = defaultValue;
     let action: string | null = null;
-    const banditKey = variation;
     try {
-      const banditParameters = this.banditConfigurationStore?.get(banditKey);
-      if (banditParameters && !Object.keys(actions ?? {}).length) {
-        // If it's a bandit, but we have no actions, just return default value
-        variation = defaultValue;
-      } else if (banditParameters) {
-        // For now, we use the shortcut of assuming if a variation value is the key of a known bandit, that is the bandit we want
+      const banditFlagAssociations = this.banditFlagConfigurationStore?.get(flagKey);
+      if (banditFlagAssociations && !Object.keys(actions).length) {
+        // No actions passed for a flag known to have an active bandit, so we just return the default values so that
+        // we don't log a variation or bandit assignment
+        return { variation, action };
+      }
+
+      // Get the assigned variation for the flag with a possible bandit
+      variation = this.getStringAssignment(flagKey, subjectKey, subjectAttributes, defaultValue);
+
+      // Check if the assigned variation is an active bandit
+      // Note: the reason for non-bandit assignments include the subject being bucketed into a non-bandit variation or
+      // a rollout having been done.
+      const banditKey = banditFlagAssociations?.find(
+        (banditFlagAssociation) => banditFlagAssociation.variationValue === variation,
+      )?.key;
```

Comment:
minor: would it make sense to index associations by variation when we prepare them? This way, this lookup will become O(1) amortized instead of O(n).

Reply:
Practically, right now we have a 1:1 mapping of flags to bandits so this will be O(1).

I'm just getting a bit ahead here where multiple flags can share a bandit and a flag can have multiple values. Then "n" in O(n) is the number of bandits a flag has, which I imagine will be very small.